openconcept/analysis/openaerostruct/tests/test_aerostructural.py

```diff
@@ -0,0 +1,411 @@
+        # Regression test
+        twist = np.array([-1, -0.5, 2])  # deg
+        toverc = np.array([0.05, 0.1, 0.12])
+        t_skin = np.array([5, 13, 15])  # mm
+        t_spar = np.array([5, 13, 15])  # mm
+        p = om.Problem(
+            OASDataGen(
+                num_x=3,
+                num_y=7,
+                num_twist=twist.size,
+                num_toverc=toverc.size,
+                num_skin=t_skin.size,
+                num_spar=t_spar.size,
+                Mach_train=np.linspace(0.1, 0.85, 2),
+                alpha_train=np.linspace(-10, 15, 2),
+                alt_train=np.linspace(0, 15e3, 2),
+            )
+        )
+        p.setup()
+        p.set_val("fltcond|TempIncrement", 0, units="degC")
+        p.set_val("ac|geom|wing|S_ref", 100, units="m**2")
+        p.set_val("ac|geom|wing|AR", 10)
+        p.set_val("ac|geom|wing|taper", 0.1)
+        p.set_val("ac|geom|wing|c4sweep", 20, units="deg")
+        p.set_val("ac|geom|wing|twist", twist, units="deg")
+        p.set_val("ac|geom|wing|toverc", toverc)
+        p.set_val("ac|geom|wing|skin_thickness", t_skin, units="mm")
+        p.set_val("ac|geom|wing|spar_thickness", t_spar, units="mm")
+        p.set_val("ac|aero|CD_nonwing", 0.01)
+        p.run_model()
+
+        # Check that the values don't change
+        CL = np.array(
+            [
+                [[-0.79243052, -0.79557334], [1.30041913, 1.30407159]],
+                [[-0.62943499, -0.76690382], [1.08568083, 1.26980739]],
+            ]
+        )
+        CD = np.array(
+            [[[0.0423459, 0.04473217], [0.07565092, 0.0781007]], [[0.03668226, 0.04308131], [0.11930142, 0.15852171]]]
+        )
+
+        assert_near_equal(p.get_val("CL_train"), CL, tolerance=1e-7)
+        assert_near_equal(p.get_val("CD_train"), CD, tolerance=1e-7)
+
+        # The fundamental problem with the derivative inaccuracy is rooted in OpenAeroStruct,
+        # not this code. There are no analytic derivatives explicitly implemented in any of
+        # the components in aerostructural.py, so the purpose for this check is just to
+        # ensure that the derivatives are assembled properly.
+        partials = p.check_partials(step=1e-7)
+        assert_check_partials(partials, atol=1e-2, rtol=1.5e-2)
+
+    def test_different_surf_options(self):
+        # Test that when there are different surf_options within a single model it catches it
+        p = om.Problem()
+        p.model.add_subsystem("one", OASDataGen(surf_options={"a": 1.13521, "b": np.linspace(0, 1, 10)}))
+        p.model.add_subsystem("two", OASDataGen(surf_options={"a": 1.13521, "b": np.linspace(0, 1, 10)}))
+        p.model.add_subsystem("three", OASDataGen(surf_options={"a": 1.13521, "b": np.linspace(0, 1, 10)}))
+        p.setup()
+
+        p = om.Problem()
+        p.model.add_subsystem("one", OASDataGen(surf_options={"a": 1.13521}))
+        p.model.add_subsystem("two", OASDataGen(surf_options={"a": 1.1352}))
```

Comment:
Why is this number (slightly) different from the rest?

Reply:
This is to check that the `OASDataGen` class is throwing an error when the user tries to setup two instances with different `surf_options` (it's not allowed because it messes with the data being shared across instances).